client/my-sites/jetpack-search/module-config.tsx

```diff
@@ -0,0 +1,105 @@
+/**
+ * External dependencies
+ */
+import React, { ReactElement, Fragment, useEffect, useState } from 'react';
+import { translate } from 'i18n-calypso';
+import { useSelector } from 'react-redux';
+
+/**
+ * Internal component dependencies
+ */
+import { Card } from '@automattic/components';
+import FormToggle from 'calypso/components/forms/form-toggle';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import SettingsSectionHeader from 'calypso/my-sites/site-settings/settings-section-header';
+import QuerySiteSettings from 'calypso/components/data/query-site-settings';
+import FormSettingExplanation from 'calypso/components/forms/form-setting-explanation';
+
+/**
+ * Internal state dependencies
+ */
+import isJetpackModuleActive from 'calypso/state/selectors/is-jetpack-module-active';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { isJetpackSite } from 'calypso/state/sites/selectors';
+import { getSiteSettings } from 'calypso/state/site-settings/selectors';
+import hasSearchProductSelector from './hooks/has-search-product';
+
+type PartialSiteSettings = { instant_search_enabled: boolean; jetpack_search_enabled: boolean };
+
+export default function JetpackSearchModuleConfig(): ReactElement {
+	const siteId = useSelector( ( state ) => getSelectedSiteId( state ) as number );
+	const siteSettings = useSelector(
+		( state ) => getSiteSettings( state, siteId ) as PartialSiteSettings
+	);
+	const hasSearchProduct = useSelector( hasSearchProductSelector );
+
+	// Server-side config values
+	const isSearchActive = useSelector( ( state ) => {
+		if ( isJetpackSite( state, siteId ) ) {
+			return isJetpackModuleActive( state, siteId, 'search' );
+		}
+
+		return siteSettings && siteSettings[ 'jetpack_search_enabled' ];
+	} );
+	const isInstantSearchActive = siteSettings && siteSettings[ 'instant_search_enabled' ];
+
+	const isRequestingSettings = false;
+	const isSavingSettings = false;
+
+	// Local UI state values
+	const [ searchEnabled, onChangeSearchEnabled ] = useState( !! isSearchActive );
+	const [ instantSearchEnabled, onChangeInstantSearchEnabled ] = useState(
+		!! isInstantSearchActive
+	);
+
+	// If server side values update, update local UI state values
+	useEffect( () => {
+		searchEnabled !== isSearchActive && onChangeSearchEnabled( !! isSearchActive );
+	}, [ searchEnabled, isSearchActive ] );
+	useEffect( () => {
+		instantSearchEnabled !== isInstantSearchActive &&
+			onChangeSearchEnabled( !! isInstantSearchActive );
+	}, [ instantSearchEnabled, isInstantSearchActive ] );
+
+	return (
+		<Fragment>
+			<QuerySiteSettings siteId={ siteId } />
+			<SettingsSectionHeader
+				title={ translate( 'General', { context: 'Settings header' } ) }
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **5** times:
`translate( 'General', { context: 'settings filter'} )` *ES Score: 13*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jetpack-search-config-panel)


:information_source: This string already exists with the following **contexts**:
- `null` (no context)
- `Help topic`
- `settings screen`

 Would it make sense to reuse one of the above?